pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1307,9 +1413,25 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 
 // evictPods queues an eviction for the provided node name, and returns false if the node is already
 // queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// If node was already evicted, the pods are deleted immediately.
+func (nc *Controller) evictPods(node *v1.Node, pods []v1.Pod) bool {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
+	if nc.nodeEvictionMap.getStatus(node.Name) == evicted {
+		// Node eviction already happened for this node.
+		// Handling immediate pod deletion.
+		remaining, err := nodeutil.DeletePods(nc.kubeClient, pods, nc.recorder, node.Name, string(node.UID), nc.daemonSetStore)
```

Comment:
What does remaining mean? What if we try to DeletePods for pods that we already tried before?

Reply:
https://github.com/kubernetes/kubernetes/blob/833e8dc10bdce43fc8229c39639a7f7364703874/pkg/controller/util/node/controller_utils.go#L42-L44

> What if we try to DeletePods for pods that we already tried before?

Good question. I think I should add special handling for the case when we try to delete pod that doesn't exist anymore.

Reply:
Handling added.